dbms/src/Storages/KVStore/MultiRaft/PreHandlingTrace.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Storages/KVStore/MultiRaft/PreHandlingTrace.h>
+
+#pragma GCC diagnostic push
+#ifdef __clang__
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+// include to suppress warnings on NO_THREAD_SAFETY_ANALYSIS. clang can't work without this include, don't know why
+#include <grpcpp/security/credentials.h>
+#pragma GCC diagnostic pop
```

Comment:
If we don't include this header, it will run into warning messages like this. Just workaround them.


```
/tiflash-master/dbms/src/Storages/KVStore/tests/gtest_read_index_worker.cpp:196:76: error: expected ';' at end of declaration list
    void operator()(std::unordered_map<RegionID, ReadIndexDataNodePtr> & d) NO_THREAD_SAFETY_ANALYSIS
                                                                           ^
                                                                           ;
//tiflash-master/dbms/src/Storages/KVStore/tests/gtest_read_index_worker.cpp:238:24: error: expected ';' at end of declaration list
            void test() NO_THREAD_SAFETY_ANALYSIS
                       ^
                       ;
2 errors generated.
```